packages/esbuild/esbuild.bzl

```diff
@@ -0,0 +1,220 @@
+            mandatory = True,
+            allow_single_file = True,
+            doc = "The bundle's entry point (e.g. your main.js or app.js or index.js)",
+        ),
+        "external": attr.string_list(
+            default = [],
+            doc = "A list of module names that are treated as external and not included in the resulting bundle",
+        ),
+        "format": attr.string(
+            values = ["iife", "cjs", "esm", ""],
+            mandatory = False,
+            doc = """The output format of the bundle, defaults to iife when platform is browser
+and cjs when platform is node. If performing code splitting, defaults to esm""",
+        ),
+        "minify": attr.bool(
+            default = False,
+            doc = """Minifies the bundle with the built in minification.
+Removes whitespace, shortens identifieres and uses equivalent but shorter syntax.
+
+Sets all --minify-* flags
+            """,
+        ),
+        "output": attr.output(
+            mandatory = False,
+            doc = "Name of the output file when bundling",
+        ),
+        "output_map": attr.output(
+            mandatory = False,
+            doc = "Name of the output source map when bundling",
+        ),
+        "platform": attr.string(
+            default = "browser",
+            values = ["node", "browser", "neutral", ""],
+            doc = "The platform to bundle for",
+        ),
+        "sources_content": attr.bool(
+            mandatory = False,
+            default = False,
+            doc = "If False, omits the `sourcesContent` field from generated source maps",
+        ),
+        "splitting": attr.bool(
+            default = False,
+            doc = """If true, esbuild produces an output directory containing all the output files from code splitting 
+            """,
+        ),
+        "srcs": attr.label_list(
+            allow_files = True,
+            default = [],
+            doc = """Non-entry point JavaScript source files from the workspace.
+
+You must not repeat file(s) passed to entry_point""",
+        ),
+        "target": attr.string(
+            default = "es2015",
+            doc = """Environment target (e.g. es2017, chrome58, firefox57, safari11, 
+edge16, node10, default esnext)
+            """,
+        ),
+        "tool": attr.label(
+            allow_single_file = True,
+            mandatory = True,
+            executable = True,
+            cfg = "exec",
+            doc = "An executable for the esbuild binary",
+        ),
+    },
+    implementation = _esbuild_impl,
+)
+
+def esbuild(name, splitting = False, **kwargs):
+    """esbuild helper macro around the `esbuild_bundle` rule
+
+    For a full list of attributes, see the `esbuild_bundle` rule
+
+    Args:
+        name: The name used for this rule and output files
+        splitting: If `True`, produce a code split bundle in an output directory
+        **kwargs: All other args from `esbuild_bundle`
+    """
+
+    if splitting == True:
```

Comment:
maybe we should make this more similar to rollup_bundle and other rules with output_dir=True ?